inc/Store/DataSource/DataSourceConfigManager.php

```diff
@@ -0,0 +1,264 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Store\DataSource;
+
+use RemoteDataBlocks\Editor\BlockManagement\ConfigStore;
+use RemoteDataBlocks\WpdbStorage\DataSourceCrud;
+use WP_Error;
+
+class DataSourceConfigManager {
+	public const CONFIG_SOURCE_CODE = 'code';
+	public const CONFIG_SOURCE_STORAGE = 'storage';
+	public const CONFIG_SOURCE_CONSTANT = 'constant';
+	public const MUTABLE_CONFIG_SOURCES = [ self::CONFIG_SOURCE_STORAGE ];
+
+	private static function get_all_from_storage(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_STORAGE ] );
+			},
+			DataSourceCrud::get_configs()
+		);
+	}
+
+	private static function get_all_from_constant(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_CONSTANT ] );
+			},
+			ConstantConfigStore::get_configs()
+		);
+	}
+
+	private static function get_all_from_code(): array {
+		return array_map(
+			function ( array $config ) {
+				return array_merge( $config, [ 'config_source' => self::CONFIG_SOURCE_CODE ] );
+			},
+			ConfigStore::get_data_sources_as_array()
+		);
+	}
+
+	/**
+	 * Quick and dirty de-duplication of data sources. If the data source does
+	 * not have a UUID (because it is registered in code), we generate an
+	 * identifier based on the display name and service name.
+	 */
+	private static function de_duplicate_configs( array $configs ): array {
+		return array_values( array_reduce(
+			$configs,
+			function ( array $acc, array $item ) {
+				$identifier = $item['uuid'] ?? md5(
+					sprintf( '%s_%s', $item['service_config']['display_name'], $item['service'] )
+				);
+				$acc[ $identifier ] = $item;
+				return $acc;
+			},
+			[]
+		) );
+	}
+
+	/**
+	 * Get all data sources from all origins with optional filters.
+	 * 
+	 * Supported filters:
+	 * - service: Filter by service name (e.g. 'airtable', 'google-sheets', 'shopify')
+	 * - enable_blocks: Filter by blocks enabled status (false matches with null/false and true matches with true)
+	 * 
+	 * Any unsupported filter keys will be ignored.
+	 * 
+	 * @param array{
+	 *   service?: string,
+	 *   enable_blocks?: bool
+	 * } $filters Optional filters to apply to the results.
+	 * @return array<array{
+	 *   uuid?: string,
+	 *   service: string,
+	 *   service_config: array<string, mixed>,
+	 *   config_source: string,
+	 *   __metadata?: array{
+	 *     created_at: string,
+	 *     updated_at: string
+	 *   }
+	 * }>
+	 */
+	public static function get_all( array $filters = [] ): array|WP_Error {
+		$code_configured = self::get_all_from_code();
+		$constant_configured = self::get_all_from_constant();
+		$storage_configured = self::get_all_from_storage();
+
+		/**
+		 * De-duplicate configs.
+		 * 
+		 * Precedence (lowest to highest):
```

Comment:
Is there an existing doc within the repo that this could be added to? If not, could one be made? This seems like an inner working of the plugin that should be easily searchable for the future. I'm thinking something like Architecture Decisions maybe?